File: app/src/main/java/com/codekk/net/NetFunc.java
package com.codekk.net;

import com.codekk.ui.base.BaseModel;

import io.reactivex.annotations.NonNull;
import io.reactivex.functions.Function;

/**
 * by y on 2017/5/16
 */

public class NetFunc<T> implements Function<BaseModel<T>, T> {

    @Override
    public T apply(@NonNull BaseModel<T> tBaseModel) {
        if (tBaseModel.getCode() != 0) {
            throw new NetException(tBaseModel.getCode(), tBaseModel.getMessage());
        }
        return tBaseModel.getData();
    }
}

File: app/src/main/java/com/codekk/net/NetFunc.kt
package com.codekk.net

import com.codekk.ui.base.BaseModel

import io.reactivex.annotations.NonNull
import io.reactivex.functions.Function

/**
 * by y on 2017/5/16
 */

class NetFunc<T> : Function<BaseModel<T>, T> {

    override fun apply(@NonNull tBaseModel: BaseModel<T>): T {
        if (tBaseModel.code != 0) {
            throw NetException(tBaseModel.code, tBaseModel.message)
        }
        return tBaseModel.data
    }
}